cedar-policy-core/src/parser/cst_to_ast.rs

```diff
@@ -834,9 +841,34 @@ enum PrincipalOrResource {
     Resource(ResourceConstraint),
 }
 
+enum TolerantAstSetting {
+    NotTolerant,
+    #[cfg(feature = "tolerant-ast")]
+    Tolerant,
+}
```

Comment:
Why are we adding an `enum` like this?

Reply:
I think this is a nice way to support both tolerant and non-tolerant ASTs even if the feature is enabled. Is there a better way to handle this that you know of?